lib/ansible/modules/cloud/amazon/elb_application_lb.py

```diff
@@ -719,14 +719,20 @@ def create_or_update_elb_listeners(connection, module, elb):
     for listener_to_add in listeners_to_add:
         try:
             listener_to_add['LoadBalancerArn'] = elb['LoadBalancerArn']
-            connection.create_listener(**listener_to_add)
+            # Rules is not a valid parameter for create_listener
```

Comment:
When a task is run with extra verbosity it should display the arg spec the module used after the 'invocation' key (run playbook with -vvv). I could see that somewhere that actual dictionary was being modified, rather than a copy of it. 
I thought I kept running the wrong playbook or was doing something wrong syntactically because it was deleting my rules:
```
...
            "listeners": [
                {
                    "DefaultActions": [
                        {
                            "TargetGroupArn": "arn:aws:elasticloadbalancing:us-east-1:448830907657:targetgroup/sherteltg/bd83c5c780de26d7",
                            "Type": "forward"
                        }
                    ],
                    "ListenerArn": "arn:aws:elasticloadbalancing:us-east-1:448830907657:listener/app/shertel-application-lb/365334adfd1f3606/9118ad2fccc38cde",
                    "Port": 80,
                    "Protocol": "HTTP",
                    "Rules": []
                }
            ],
```